tests/framework/extensions/nodes/nodes.go

```diff
@@ -0,0 +1,66 @@
+package nodes
+
+import (
+	"context"
+
+	"github.com/rancher/rancher/pkg/api/scheme"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var NodeGroupVersionResource = schema.GroupVersionResource{
+	Group:    "",
+	Version:  "v1",
+	Resource: "nodes",
+}
+
+// get the nodes
+func GetNodes(client *rancher.Client, clusterID string, listOpts metav1.ListOptions) ([]corev1.Node, []error) {
+	var nodesList []corev1.Node
+
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	nodeResource := dynamicClient.Resource(NodeGroupVersionResource)
+	nodes, err := nodeResource.List(context.TODO(), listOpts)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	for _, unstructuredNode := range nodes.Items {
+		newNode := &corev1.Node{}
+		err := scheme.Scheme.Convert(&unstructuredNode, newNode, unstructuredNode.GroupVersionKind())
+		if err != nil {
+			return nil, []error{err}
```

Comment:
Where are you returning a list of errors. Instead if an error is detected return it. No need to get a list of them.